docs/custom_installation.md

```diff
@@ -0,0 +1,56 @@
+## Custom Installation
+--------------------
+
+__If you are installing this module using a control-repo, you must have splunk_hec in your production environment's Puppetfile so the puppetserver process can load the libraries it needs properly. You can then create a feature branch to enable them and test the configuration, but the libraries must be in production otherwise the feature branch wont work as expected.__
+
+The steps below will help install and troubleshoot the report processor on a single Puppet Master, including manual steps to configure a puppet-server, and to use the included splunk_hec class. Because one is modifying production machines, these steps allow you to validate your settings before deploying the changes live. See the tl,dr; instructions for 
+
+1. Install the Puppet Report Viewer Addon in Splunk. This will import the needed sourcetypes to configure Splunk's HTTP Endpoint Collector (HEC) and provide a dashboard that will show the reports once they are sent to Splunk.
+
+2. Create a Splunk HEC Token or use an existing one that sends to main index and does not have acknowledgement enabled. Follow the steps provided by Splunk's [Getting Data In Guide](http://docs.splunk.com/Documentation/Splunk/latest/Data/UsetheHTTPEventCollector) if you are new to HTTP Endpoint Collectors.
+
+3. Install this Puppet module in the environment that manages your Puppet Servers are using (probably `production`)
+
+4. Run `puppet plugin download` on your Puppet Server to sync the content
```

Comment:
We have some reports that the permissions of the plugin sync'ed files may be causing issues if they have a restrive `umask`. This only happens with `puppetserver` as it is run with the `pe-puppet` user and may not have read/execute on those files. Maybe we could have them run `(umask 022 && puppet plugin download)`

Reply:
We can definitely add a note suggesting users run `umask 022`

Reply:
A note about the umask would be great. An addition to the troubleshooting section would help for people that are hitting this issue.